src/colmap/scene/reconstruction.cc

```diff
@@ -43,6 +43,33 @@ namespace colmap {
 
 Reconstruction::Reconstruction() : max_point3D_id_(0) {}
 
+Reconstruction::Reconstruction(const Reconstruction& recon)
```

Comment:
Would it make sense to add a test for these two new methods and assert equality between the two objects (or their serialized states)?

Reply:
Sorry maybe I am missing something. The Reconstruction class in colmap does not naturally support serialization? Do we write a new method for this?

Reply:
You can read and write the reconstruction using reconstruction_io?

Reply:
Thanks. This has been added. Wrote a new function for comparing binary files. I hope I did not miss any existing utilities in the codebase. PTAL

Reply:
I created a new PR that should hopefully simplify the tests in this PR: https://github.com/colmap/colmap/pull/2842